### PR TITLE
Commands System and Logging (chandler branch)

### PR DESCRIPTION
**Pull request by chandlerwr**
Addresses #15 and #14 as a possible solution.
Commands system broken into an object in its own file (prevents clutter in players.js).
Logging system also in its own file. Used via: `var logger = require("./logger.js").logger` and `logger.info("message")`.

### CODE REVIEW

File: src/cmds.js [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                    }
+
+                    // display the command usage
+                } else sendUsage(args[0], message);
+                
+
+                // help with no argument
+            } else {
+
+                let msg = "Commands:"
```
Comment (ben-dow) [inline src/cmds.js]: Missing Semicolon

File: src/cmds.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+const commands = {
+
+    help: {
+        name: 'help',
+        usage: config.prefix + 'help [command]',
+        help: 'displays the command list with command summaries',
+
+        // args should be string[], message should be the discord Message as given in the message event
+        func: function (args, message) {
+
```
Comment (ben-dow) [inline src/cmds.js:18]: As this file is likely to get very long, lets move Functionality out of this file and trigger the functionality from here. Keeping command functionality out of this file will keep it from getting absurdly long

File: src/cmds.js [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+            }
+        }
+    },
+
+    ttmtg: {
+        name: "ttmtg",
+        usage: config.prefix + 'ttmtg',
+        help: 'displays the time until the next RIT Players meeting',
+        func: function (args, message) {
+
```
Comment (ben-dow) [inline src/cmds.js]: Same issue here about Moving functionality out of this file

File: src/cmds.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    },
+
+    remind: {
+
+        name: "remind",
+        usage: config.prefix + 'remind ()', // todo
+        help: 'add a reminder to this server',
+        func: function (args, message) {
+
+            logCommand(this.name, args); // logging
```
Comment (ben-dow) [inline src/cmds.js]: Placement of Logging in this file is good and logical

File: src/cmds.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+
+/**
+ * To be run when an unknown command sent in a message.
+ * @param {string} cmd      the command
+ * @param {string[]} args   the command arguments
+ * @param {Message} message the message that the command was sent in
+ */
+function unknownCommand(cmd, args, message) {
+
```
Comment (ben-dow) [inline src/cmds.js]: Same Comment about moving functionality. For these unknown command, sendUsage  and LogCommand may be good to make a file containing the word "utility" to indicate commands and responses that will be used, but not necessarily triggered directly by a user.

File: src/cmds.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+}
+
+
+/**
+ * Respond to a message with the usage of a command.
+ * @param {string} cmd      the command
+ * @param {Message} message the message to respond to
+ */
+function sendUsage(cmd, message) {
+
```
Comment (ben-dow) [inline src/cmds.js]: If this is only being used in the Help Function, may want to move it as well into the file for functionality for the help function

**Review by ben-dow — CHANGES_REQUESTED** [review]
Comments made on individual files. Overall looks good, but to prep, for the cmd.js file to get absurdly long, should be moving functionality (where it makes sense) to separate files for readability sake

**Comment by ben-dow** [comment]
Not sure if possible to change, but pull requests, in general, should be made to dev instead of master

File: src/cmds.js (hunk 6 above)
Comment (chandlerwr) [reply src/cmds.js]: Other functions will be using `sendUsage` if they are used incorrectly.

**Pull request merged**